From 31b76caddc82ade7c46fd39fccdf3f753dff759b Mon Sep 17 00:00:00 2001
From: Drew Moseley <drew.moseley@northern.tech>
Date: Mon, 20 May 2019 23:17:44 +0000
Subject: [PATCH] mx6var_som: Find splash screen on active rootfs with Mender.

Note that this patch is hardcoded to the Mender defines that are
expected. Given when this code executes, the logic in mender_setup has
not run yet so we are just hard coding the values here.

Signed-off-by: Drew Moseley <drew.moseley@northern.tech>
---
 board/variscite/mx6var_som/mx6var_som.c | 26 +++++++++++++++----------
 include/configs/mx6var_som.h            |  1 -
 2 files changed, 16 insertions(+), 11 deletions(-)

diff --git a/board/variscite/mx6var_som/mx6var_som.c b/board/variscite/mx6var_som/mx6var_som.c
index 45dbb8e0d8..11dfab0335 100644
--- a/board/variscite/mx6var_som/mx6var_som.c
+++ b/board/variscite/mx6var_som/mx6var_som.c
@@ -194,15 +194,15 @@ static int env_check(char *var, char *val)
 #ifdef CONFIG_SPLASH_SCREEN
 static void set_splashsource_to_boot_rootfs(void)
 {
-	if (!env_check("splashsourceauto", "yes"))
-		return;
-
 #ifdef CONFIG_NAND_BOOT
-	env_set("splashsource", env_get("rootfs_device"));
+	if (env_check("mender_uboot_if", "mmc"))
+		env_set("splashsource", "emmc");
+        else
+		env_set("splashsource", "nand");
 #else
-	if (mmc_get_env_dev() == 0)
+	if (env_check("mender_uboot_dev", "0"))
 		env_set("splashsource", "sd");
-	else if (mmc_get_env_dev() == 1)
+	else
 		env_set("splashsource", "emmc");
 #endif
 }
@@ -213,12 +213,18 @@ int splash_screen_prepare(void)
 
 	char sd_devpart_str[5];
 	char emmc_devpart_str[5];
+	char ubi_devpart_str[15], ubi_part;
 	u32 sd_part, emmc_part;
 
-	sd_part = emmc_part = env_get_ulong("mmcrootpart", 10, 0);
+	sd_part = emmc_part = env_get_ulong("mender_boot_part", 10, 0);
+	if (sd_part == 0)
+		ubi_part = 'a';
+	else
+		ubi_part = 'b';
 
-	sprintf(sd_devpart_str, "0:%d", sd_part);
-	sprintf(emmc_devpart_str, "1:%d", emmc_part);
+	snprintf(sd_devpart_str, 5, "0:%d", sd_part);
+	snprintf(emmc_devpart_str, 5, "1:%d", emmc_part);
+	snprintf(ubi_devpart_str, 15, "ubi0:rootfs%c", ubi_part);
 
 	struct splash_location var_splash_locations[] = {
 		{
@@ -238,7 +244,7 @@ int splash_screen_prepare(void)
 			.storage = SPLASH_STORAGE_NAND,
 			.flags = SPLASH_STORAGE_FS,
 			.mtdpart = "rootfs",
-			.ubivol = "ubi0:rootfs",
+			.ubivol = ubi_devpart_str,
 		},
 	};
 
diff --git a/include/configs/mx6var_som.h b/include/configs/mx6var_som.h
index 074012d9dd..b1cc007656 100644
--- a/include/configs/mx6var_som.h
+++ b/include/configs/mx6var_som.h
@@ -117,7 +117,6 @@
 	"fdt_file=undefined\0" \
 	"fdt_addr=0x18000000\0" \
 	"fdt_high=0xffffffff\0" \
-	"splashsourceauto=yes\0" \
 	"splashfile=/boot/splash.bmp\0" \
 	"splashimage=0x18100000\0" \
 	"splashenable=setenv splashfile /boot/splash.bmp; " \
-- 
2.21.0

